### ISSUE TITLE
counsel-projectile-grep shows 1 chars more

### ISSUE THREAD

**Issue by bluewww** [ericdanan/counsel-projectile#109]
counsel-projectile-grep shows the message "1 chars more" even though more than 3 chars are already typed (and if the text doesn't occur at all in the searched files). Previousely the "1 chars more" text would disappear after the third character and the minibuffer would remain empty (since we area searching for something that doesn't occur).
edit: actually sometimes the search doesn't start at all even though a sufficient number of characters have been typed, independent of whether the search string appears somewhere in the searched files.

**Comment by ericdanan** [comment]
It works correctly for me: as soon as I have typed a 3-character string that doesn't match, the minibuffer correctly shows "No matches found". Are you running the latest versions of ivy/counsel, projectile, and counsel-projectile? Did you customize these packages? Does it happen for different types of projects (e.g. git vs non-git projects)?

**Comment by wedens** [comment]
@bluewww Try disabling your shell configuration (`~/.zshrc`, `~/.zshenv`, `~/.bashrc` or whatever you use) and see if it works.

**Comment by ericdanan** [comment]
Closing, @bluewww please reopen if needed.

**Comment by bottleneko** [comment]
It happens when used standard MacOS grep

Solve of this problem is installing `gnugrep` from brew: `brew install grep` and put this in your `.bashrc`/`.zshrc`

```
export PATH="/usr/local/opt/grep/libexec/gnubin:$PATH"
```

Don't forget source your `.*rc` file. Good luck

**Comment by ericdanan** [comment]
Thanks @bottleneko.

@bluewww if you're still using counsel-projectile, it would be interesting to know if you're indeed on MacOS and if so whether that fixes the issue for you. I can't test myself because I'm not on MacOS and I use GNU make.

**Comment by bluewww** [comment]
I only use gnu/linux systems. iirc the bug occured for me on centos7 and archlinux. Let me try again.

**Comment by bottleneko** [comment]
@bluewww also you may test your grep by 

```sh
grep -rnEI --exclude=TAGS --exclude-dir=.idea/ --exclude-dir=.git/ --exclude-dir=.hg/ --exclude-dir=.fslckout/ --exclude-dir=_FOSSIL_/ --exclude-dir=.bzr/ --exclude-dir=_darcs/ --exclude-dir=.tox/ --exclude-dir=.svn/ -- "some text for grep"
```

My output for "test" look like this

```
_checkouts/nodefinder/doc/Makefile.in:466:      -test -z "$(MAINTAINERCLEANFILES)" || rm -f $(MAINTAINERCLEANFILES)
_checkouts/nodefinder/doc/Makefile.in:583:      python -c "import sys; filename = sys.argv[1]; file = open(filename).read (); compile(file, 'test', 'exec')" "$*"
_checkouts/nodefinder/doc/Makefile.in:622:        if test $$? = 2 ;                                                  \
_checkouts/nodefinder/rebar.config:8:%% Options for eunit:test()
apps/decs/test/decs_partition_tests.erl:1:-module(decs_partition_tests).
apps/decs/test/decs_partition_tests.erl:7:chunks_test_() ->
apps/decs/priv/riak_core.schema:113:%% a developer is testing a high number of vnodes and/or several VMs
apps/decs/priv/riak_core.schema:275:%% http://docs.basho.com/riak/latest/dev/advanced/cap-controls/
apps/decs/priv/riak_core.schema:721:%% There are pro's and con's for both aproaches, it is best to test out what
```

**Comment by yoricksijsling** [comment]
I was having this problem with both `counsel-projectile-rg` and `counsel-projectile-ag`, and this issue is the first result I found.

It turned out that I just needed to wait longer. The search was really slow because I had a bunch of test data in my project directory, so I ended up just removing the data and it worked fine again.

It might be useful to have some indication that the search is still in progress.
